src/users/index.js

```diff
@@ -0,0 +1,217 @@
+const express = require('express')
+const logError = require('../logging/log_error')
+
+class User {
+  /**
+    * Make a new user with the given id
+    * @param {string} id
+    */
+  constructor (id) {
+    /**
+     * The userId for this user. This should be unique between all users, and can be used to fetch this user form the UserProvider
+     */
+    this.id = id
+  }
+
+  /**
+  * Specify that the given user would like to work on the given tasks.
+  * This is not expected to check if the taskId belongs to a valid tasks.
+  *
+  *
+  * @param {string[]} taskIds - The ids of the task which will be claimed by this user.
+   * @returns { PromiseLike<void> | void }
+  * @abstract
+  */
+  claimTasks (taskIds) { throw new Error('unimplemented') }
+
+  /**
+   * Specify that the given user would like to stop working on the given tasks.
+   * This is not expected to check if the taskId belongs to a valid tasks.
+   *
+   * If the user is not already working on the given task, then this should succeeded silently
+   *
+   * If the given user does not exist, then this should return false. Otherwise it should return true
+   *
+   * @param { string[] } taskIds - The ids of the tasks which will be unclaimed by this user.
+   * @returns { PromiseLike<void> | void }
+   * @abstract
+   */
+  unclaimTasks (taskIds) { throw new Error('unimplemented') }
```

Comment:
Do we have a need for more than the `claimTasks` method right now?

Reply:
Not strictly no. However not having it leaves the API feeling asymmetric - personally I find it very frustrating when an API provides a `foo` operation without an `unfoo` operation. 

On the other hand, it would add complexity that we don't need right now, and is unlikely to be difficult to implement in the future, so I am ok either way.

Reply:
I'm also unsure of the "business" implications as well, like whether Jerry would receive emails on unclaiming, whether we would need to prevent unclaiming tasks that have more than zero hours volunteered, etc.

Maybe we can remove this until we know if and how unclaiming would behave?

Reply:
Hm, that's a good point. I'll go ahead and remove it